### PR TITLE
Cypress tests update

### LINKED ISSUE #24: As a user I should see error messages when things do not go as planned
- [ ] Error message when needs do not load to NeedList
- [ ] Error message when a user attempts to create a need but it cannot be submitted

### LINKED ISSUE #41: Cypress tests should be fully updated to reflect Phase 1 & Phase 2 changes and features

### LINKED ISSUE #42: As a user, I should not be able to see a person's email on the need card or need detail page

### PR DESCRIPTION
**Pull request by novaraptur**
# Description

Updated Cypress tests to reflect new functionality and more thoroughly test user flows. Added additional error handling for invalid form inputs like bad email addresses. Added redirect route for garbage URLs. Removed visible email from need cards.

Closes #42 Closes #41 Closes #24

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Refactoring

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Developer Tools
- [ ] Live Site
- [X] Cypress

## Checklist:

- [X] My code follows Turing's guidelines for this project
- [X] I have performed a self-review of my own code
- [X] I have generated no new error messages
- [X] I have rebased and merged my code

### CODE REVIEW

File: cypress/integration/help-request.js [hunk 1]
```diff
@@ -25,11 +25,94 @@ describe('Barn Raiser Help Request Form', () => {
...
+    cy.get('input[name="email"]').type('test.email@gmail.com');
+    cy.get('input[name="zipCode"]').type('80230');
+    cy.get('input[name="needDate"]').type('2021-09-15');
+    cy.get('input[name="startTime"]').type('09:00:00');
+    cy.get('input[name="endTime"]').type('08:00:00');
+    cy.get('input[name="volunteersNeeded"]').type('300');
+    cy.get('input[name="needTitle"]').type('Make a Video Game');
+    cy.get('input[name="needDescription"]').type('I need people to help me make a video game!');
+    cy.get('button').contains('Submit').click();
+    cy.get('h3').contains('Warning: Your submission could not go through.').should('be.visible');
```
Comment (aliyamerali) [inline cypress/integration/help-request.js:90]: Love this edge case testing!

File: cypress/integration/help-request.js [hunk 2]
```diff
@@ -25,11 +25,94 @@ describe('Barn Raiser Help Request Form', () => {
...
+    cy.get('input[name="startTime"]').type('09:00:00');
+    cy.get('input[name="endTime"]').type('10:00:00');
+    cy.get('input[name="volunteersNeeded"]').type('300');
+    cy.get('input[name="needTitle"]').type('Make a Video Game');
+    cy.get('input[name="needDescription"]').type('I need people to help me make a video game!');
+    cy.get('button').contains('Submit').click();
+    cy.get('h3').contains('Warning: Your submission could not go through.').should('be.visible');
+  });
+
+  it('should not allow the user to submit if the zip code is in an invalid format', () => {
```
Comment (aliyamerali) [inline cypress/integration/help-request.js:106]: Good catch to test for this on FE. It's not currently restricted on the BE/DB side, maybe we can explore a check there too.

**Review by aliyamerali — APPROVED** [review]
Great to see, thanks for tagging me! I appreciate all the edge cases you're testing here.

File: src/components/App/App.js [hunk 3]
```diff
@@ -101,6 +101,13 @@ function App() {
               </section>
               )
             }}/>
+            <Route
+              render={() => {
+                return (
+                  <Redirect to="/" />
```
Comment (ShaunaMyers) [inline src/components/App/App.js:107]: I've not used redirect before. Only switch...in the garbage url scenario. I'll have to research this more.

File: src/components/NeedForm/NeedForm.js [hunk 4]
```diff
@@ -83,11 +83,16 @@ const NeedForm = () => {
             error = true
             return;
           }
+        } else if (key === "pointOfContact") {
+          if (!variables[key].includes('@') || !variables[key].includes('.')) {
```
Comment (ShaunaMyers) [inline src/components/NeedForm/NeedForm.js:87]: This is a great addition. Although, I thought input type email gave them a warning and wouldn't let them continue without the proper email formatting. Am I wrong in assuming that?

File: cypress/integration/errors.js [hunk 5]
```diff
@@ -1,7 +1,11 @@
 describe('Barn Raiser Error Handling', () => {
-  it('should display an error page to the user and ask them to re-route back home if they go to a non-existent page');
-
-  it('should notify the user if their help request cannot go through to the server');
-
-  it('should display a message to the user if it cannot fetch help requests from the server');
+  it('should re-route the user back home if they go to a non-existent page', () => {
+    cy.visit('http://localhost:3000/garbage_url_test');
+    cy.url().should('eq', 'http://localhost:3000/');
```
Comment (ShaunaMyers) [inline cypress/integration/errors.js:4]: This was a great addition! Thanks for adding this!

**Pull request merged**